streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/StatefulProcessorNode.java

```diff
@@ -34,13 +34,24 @@
     public StatefulProcessorNode(final String nodeName,
                                  final ProcessorParameters processorParameters,
                                  final String[] storeNames,
-                                 final StoreBuilder<? extends StateStore> materializedKTableStoreBuilder,
                                  final boolean repartitionRequired) {
         super(nodeName,
-            processorParameters,
-            repartitionRequired);
+              processorParameters,
+              repartitionRequired);
 
         this.storeNames = storeNames;
+        this.storeBuilder = null;
+    }
+
+    public StatefulProcessorNode(final String nodeName,
```

Comment:
I agree having two separate constructors clears usage of `StatefulProcessorNode` up.  

Maybe we could add some Javadoc comments that the constructor taking the `StateStore` parameter is for stateful operations that create a state store.  
And the constructor taking `storeNames` is for stateful DSL methods `process`, `transform` and the other transform variants where the state store is already created (with a `StreamsBuilder.addStateStore` call) , and we pass in the names so the `Processor` can access the stores.